integrations/docker/images/stage-1/chip-build-crosscompile/start-sysroot-vm.sh

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env bash
+
+# This script is intended to build a SYSROOT that can be used as a base to cross-compile
+# CHIP for aarch64 (i.e. that runs on a RaspberryPi) from a amd64 linux machine
+
+set -e
+
+start_console=1
+
+while getopts "hin" opt; do
+    case $opt in
+        h)
+            echo "Usage: $0 [-h] [-i]"
+            echo "    -h    Displays this help message"
+            echo "    -i    Attempt to 'sudo apt-get install' required packages"
+            echo "    -n    Do not start a console after vm is created"
+            exit 0
+            ;;
+        i)
+            echo "Installing required packages ..."
+            sudo apt-get install \
+                genisoimage \
+                libvirt-daemon \
+                qemu-system-arm \
+                qemu-utils \
+                virtinst \
+                whois \
+                ;
+            ;;
+        n)
+            echo "Console will not be started after virtual machine startup."
+            start_console=0
+            ;;
+    esac
+done
+
+if ! which qemu-system-aarch64 >/dev/null; then
+    echo "Cannot find 'qemu-system-aarch64'. Did you 'sudo apt-get install qemu-system-arm' ?"
+    exit 1
+fi
+
+if ! which virt-install >/dev/null; then
+    echo "Cannot find 'virt-install'. Did you 'sudo apt-get install virtinst' ?"
+    exit 1
+fi
+
+if ! [ -f /usr/sbin/libvirtd ]; then
+    echo "Cannot find 'virt-install'. Did you 'sudo apt-get install libvirt-daemon' ?"
+    exit 1
+fi
+
+if ! which mkpasswd >/dev/null; then
+    echo "Cannot find 'mkpasswd'. Did you 'sudo apt-get install whois' ?"
+    exit 1
+fi
+
+if ! which qemu-img >/dev/null; then
+    echo "Cannot find 'qemu-img'. Did you 'sudo apt-get install qemu-utis' ?"
+    exit 1
+fi
+
+if ! which genisoimage >/dev/null; then
+    echo "Cannot find 'genisoimage'. Did you 'sudo apt-get install genisoimage' ?"
+    exit 1
+fi
+
+CLOUD_IMAGE_URL="https://cloud-images.ubuntu.com/releases/24.04/release-20240821/ubuntu-24.04-server-cloudimg-arm64.img"
+CLOUD_IMAGE_FILE=$(basename "$CLOUD_IMAGE_URL")
+
+if ! [ -f "$CLOUD_IMAGE_FILE" ]; then
+    echo "Image $CLOUD_IMAGE_FILE does not exist. Downloading ..."
+    wget "$CLOUD_IMAGE_URL"
+else
+    echo "Using existing $CLOUD_IMAGE_FILE ..."
+fi
+
+rm -f sysrootsrv.img
+qemu-img create -b "$CLOUD_IMAGE_FILE" -f qcow2 -F qcow2 sysrootsrv.img 10G
+
+# local user allowed to SSH in to the host directly without a password
+SSH_KEY=$(cat ~/.ssh/id_rsa.pub)
+
+# set a login password
+PASSWORD_HASH=$(mkpasswd 1234)
```

Comment:
i'm guessing the security story here is "ubuntu running in qemu only allows connections from `localhost`"?  is it worth noting that strategy here so folks have some peace of mind when they see a weak password here?

Reply:
We use `user` networking, so you cannot even connect to it from anywhere except through either the virtual console (which is what I added here) or through the `virsh console` which is local console only. The vm IP is not routable even from the host machine amazingly enough.

I will add a comment, however realistically this is a password so I can login from the console to debug things.

Reply:
Added text explaining password usage.